testing/test_assertion.py

```diff
@@ -677,8 +695,13 @@ def test_dict_different_items(self) -> None:
             "Right contains 2 more items:",
             "{'b': 1, 'c': 2}",
             "Full diff:",
-            "- {'b': 1, 'c': 2}",
```

Comment:
Indeed for very small diffs, the change makes it harder to read, but those are not problematic anyways, the problematic diffs are the long ones, which are greatly improved, so I think the trade-off is valid. 👍

Reply:
We could also first do a diff with the normal pprint, and if both entries are single line, use it, otherwise uses the long one if that's preferred

Reply:
Wouldn't that complicate the code quite a bit? If not I would go for it.

Reply:
What I would have in mind is something like https://github.com/pytest-dev/pytest/commit/0850f34952d697f429a37cc69d3315f4104a9969

I haven't pushed it yet to this branch as I am not convinced about it for the following reasons:

- We lose the ability of having a stable diff across python versions (since the pprint code in the standard library changes)
- Older versions of python thus don't get some of the improvements (e.g. python3.8 and dataclasses)
- The diff looks nicer for simple cases (a same length value changed), but looks worse for cases where something is missing, see [here](https://github.com/pytest-dev/pytest/commit/0850f34952d697f429a37cc69d3315f4104a9969#diff-44ed7959db9c852d42a2fd6f9b476e6f27ac670cc481f1eb59976e608e738c2aR100-R103)

As such I think I personally lean towards keeping the diff consistent even if it gets over multiple lines all the time. I'm happy to push this commit here if you prefer. In the end, the really hard to read diffs are the longer ones :)